crawler/helpers.py

```diff
@@ -97,41 +109,65 @@ def check_for_required_fields(csvreader: DictReader, centre: Dict[str, str]) ->
     Returns:
         List[str] -- list of errors experienced
     """
-    logger.debug("Checking data for required fields")
-
-    if csvreader.fieldnames and centre["barcode_field"] not in csvreader.fieldnames:
-        error = "Barcode field not in CSV file"
-        logger.error(error)
-        return [error]
+    """Checks that the CSV file has the required headers.
 
-    return []
+    Raises:
+        CentreFileError: Raised when the required fields are not found in the file
+    """
+    logger.debug("Checking CSV for required headers")
+    required_fields = {
+        FIELD_ROOT_SAMPLE_ID,
+        FIELD_RNA_ID,
+        FIELD_RESULT,
+        FIELD_DATE_TESTED,
+        FIELD_LAB_ID,
+    }
+    if csvreader.fieldnames:
+        fieldnames = set(csvreader.fieldnames)
+        if not required_fields < fieldnames:
```

Comment:
Should this be required_fields <= fieldnames? Otherwise it'll error if the two fieldsets are the same, but the actual error message then generated will be confusing. (I'm assuming it would say ' missing in CSV file'). I figure you'll almost always have additional fields anyway, so perhaps this is just a technicality.

Reply:
Yes, good spot thanks!